### PR TITLE
Nostr Wallet Connect/Dropdown Menu 

### PR DESCRIPTION
**Pull request by TheWildHustle**
**Title:** Add Nostr Wallet Connect and Dropdown Menu

**Description:** This pull request introduces two major features to the project:

1. **Nostr Wallet Connect**: Integration of Nostr wallet connectivity.
2. **Dropdown Menu**: Addition of a dropdown menu for enhanced navigation.

**Files Updated:**

- `public/index.html`: Updated to include the dropdown menu and wallet connect elements.
- `public/script.js`: Updated to handle the new dropdown menu and wallet connect functionalities.

**Files Added:**

- `src/wallet.js`: JavaScript implementation for wallet connectivity.
- `src/wallet.ts`: TypeScript definitions for wallet connectivity.
- `src/nip47.ts`: TypeScript definitions for NIP-47 protocol.
- `src/nip47.js`: JavaScript implementation for NIP-47 protocol.
- `src/app.ts`: TypeScript definitions for the main application logic.
- `src/app.js`: JavaScript implementation for the main application logic.

**Summary of Changes:**

- **Nostr Wallet Connect**:
    - Added wallet.js and wallet.ts for handling wallet connections.
    - Integrated NIP-47 protocol with nip47.js and nip47.ts.
- **Dropdown Menu**:
    - Updated index.html to include the dropdown menu structure.
    - Enhanced script.js to manage dropdown menu interactions.

Zaplist 1
Zaplist 2
Zaplist 3

**Testing:**

- Ensure that the dropdown menu is functional and responsive.
- Verify that the Nostr wallet connect feature works as expected.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -16,8 +18,27 @@
     <script src="https://cdn.jsdelivr.net/npm/flatpickr"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/jszip/3.7.1/jszip.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/FileSaver.js/2.0.5/FileSaver.min.js"></script>
+    <script type="module" src="https://esm.sh/@getalby/bitcoin-connect@3.6.2"></script>
     <link rel="preload" href="https://image.nostr.build/56795451a7e9935992b6078f0ee40ea4b0013f8efdf954fb41a3a6a7c33f25a7.png" as="image">
+    
     <style>
+
```
Comment (bitkarrot) [inline public/index.html]: lets move the style to a separate CSS file

File: public/script.js
Comment (bitkarrot) [inline public/script.js:1]: Lets rename script.js to something that is more descriptive to its function, for example if the logic is only for dropdown menu, then name the file to "menu.js" , and if there are other functions related to  wallet logic or nostr logic, split into a separate file and name accordingly. this way as a project grows in function, it  is easier to manage and clear

File: src/wallet.ts
Comment (bitkarrot) [inline src/wallet.ts:1]: I am not sure why you have duplicate .ts and .js unless this is meant as practice?

**Review by bitkarrot — CHANGES_REQUESTED** [review]
please make changes as per prior comments. thanks

**Comment by bitkarrot** [comment]
how's it going ?

**Comment by TheWildHustle** [comment]
Broke wallet and login button functionality when moving files and folders around. 
But it looks like the imports have the correct routes.... 
I'll try to have it completed by tomorrow.

**Comment by bitkarrot** [comment]
One of the things i noticed is the usage of custom css instead of the pico css framework. It might help to stick closer to the framework for some items so as less to modify and maintain. 

Also try separating and isolating out sections so that they are easy to comment out and you are able to focus on a new feature instead of carrying the weight of everything in the repo. Its a common stumbling block for new devs, which makes it harder to keep track of things as more is added and more technical debt is created.

**Pull request closed without merging**